### PR TITLE
Fix cli bugs

### PR DESCRIPTION
**Pull request by evanxg852000**
### Context / purpose
There are still lot of small issues on the cli .
- https://github.com/quickwit-inc/quickwit/issues/90
- https://github.com/quickwit-inc/quickwit/issues/92
- https://github.com/quickwit-inc/quickwit/issues/95
- https://github.com/quickwit-inc/quickwit/issues/96

### Description
- Non-existent file piped into the cli is fixed (need to test on other os)
- Fixed merge before building hotcache
- Fixed cli command ran in non-interractive mode with piped input. (need to run on windows for EOF testing)

### How was this PR tested?
Running the various command with all sorts of input on (win, mac, ubuntu)

### Checklist
- [x] tested locally
- [ ] added unit tests
- [ ] included documentation

### CODE REVIEW

**Comment by fmassot** [comment]
When executing action create / index / delete, the index fold is still here.

**Comment by evanxg852000** [comment]
@fmassot  the delete is now handled in https://github.com/quickwit-inc/quickwit/pull/102 as it was not implemented.

**Comment by fulmicoton** [comment]
Can we solve at least #96 separately?
It is quite urgent since nothing works without solving it.

File: quickwit-core/src/indexing/index.rs [hunk 1]
```diff
@@ -45,8 +46,10 @@ const SPLIT_CHANNEL_SIZE: usize = 30;
 pub struct IndexDataParams {
     /// Index uri.
     pub index_uri: PathBuf,
-    /// Input path from where to read new-line delimited json documents
+    /// Input path from where to read new-line delimited json documents.
     pub input_uri: Option<PathBuf>,
+    /// Denotes if the StdIn is running in interractive mode.
+    pub is_stdin_atty: bool,
```
Comment (fulmicoton) [inline quickwit-core/src/indexing/index.rs:52]: after rebase you can remove that... see other comment

File: quickwit-core/src/indexing/index.rs [hunk 2]
```diff
@@ -89,8 +92,13 @@ pub async fn index_data(
         .await?;
     }
 
-    if params.input_uri.is_none() {
-        println!("Please enter your new line delimited json documents.");
```
Comment (fulmicoton) [inline quickwit-core/src/indexing/index.rs:93]: after rebase this logic happens in the main.rs file, so you don't need to put any printing or stdout logic in here.
(as always separate the concern)

**Comment by fulmicoton** [comment]
LGTM jsut need to rebase.

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**